feat(terraform): add conditional logic for database configuration

sachincool opened the pull request:
Introduce conditional logic to enable or disable database
configurations based on `truefoundry_db_enable` variable. This
enhancement allows for more flexible deployment scenarios by
conditionally setting backup, IP configuration, and root password
only when the database is enabled. Additionally, add validation
for `truefoundry_db_network_cidr` and `truefoundry_db_zone` to
ensure valid inputs when the database is enabled.

sachincool committed:
feat(terraform): add conditional logic for database configuration

Introduce conditional logic to enable or disable database
configurations based on `truefoundry_db_enable` variable. This
enhancement allows for more flexible deployment scenarios by
conditionally setting backup, IP configuration, and root password
only when the database is enabled. Additionally, add validation
for `truefoundry_db_network_cidr` and `truefoundry_db_zone` to
ensure valid inputs when the database is enabled.

github-actions[bot] committed:
terraform-docs: automated action

dunefro commented on postgres.tf:
```diff
@@ -18,15 +17,16 @@ module "postgresql-db" {
   availability_type = var.truefoundry_db_availability_type
 
-  backup_configuration = {
+  backup_configuration = var.truefoundry_db_enable ? {
```
This is not required, we are already applying a count block

dunefro commented on variables.tf:
```diff
@@ -51,7 +51,19 @@ variable "truefoundry_db_postgres_version" {
   type        = string
-  description = "Network CIDR for the truefoundry postgres database. Minimum range is /24. This CIDR must be different from the main subnet where your cluster is gettin created. This subnet is created inside the GCP's network."
+  description = "Network CIDR for the truefoundry postgres database. Minimum range is /24. This CIDR must be different from the main subnet where your cluster is getting created. This subnet is created inside the GCP's network."
+  default     = "10.0.0.0/24"
```
Please don't add a default. Why are we adding a default here ?

dunefro replied:
Default value should be empty string

dunefro commented on variables.tf:
```diff
@@ -51,7 +51,19 @@ variable "truefoundry_db_postgres_version" {
+      var.truefoundry_db_enable == false ? true : (
+        var.truefoundry_db_network_cidr != null &&
+        can(regex("^([0-9]{1,3}\\.){3}[0-9]{1,3}/[0-9]{1,2}$", var.truefoundry_db_network_cidr)) &&
+        tonumber(split("/", var.truefoundry_db_network_cidr)[1]) <= 24
```
`/24` is preferred but is a range of 256 IP address which is a lot wrt a DB, please keep it `/28`

dunefro commented on variables.tf:
```diff
@@ -116,6 +128,12 @@ variable "truefoundry_db_tier" {
+  default     = null
+
+  validation {
+    condition     = var.truefoundry_db_enable == false ? true : var.truefoundry_db_zone != null
```
```suggestion
    condition     = var.truefoundry_db_enable ? var.truefoundry_db_zone != null: true
```

dunefro commented on variables.tf:
```diff
@@ -51,7 +51,19 @@ variable "truefoundry_db_postgres_version" {
+
+  validation {
+    condition = (
+      var.truefoundry_db_enable == false ? true : (
```
Please make the condition as `var.truefoundry_db_enable. ? <> : <>`

sachincool committed:
refactor(terraform): update database configuration and validation logic

Modify database configuration to always enable backups and adjust network CIDR validation to allow smaller subnet masks. Simplify conditional logic for database-related variables to improve clarity and flexibility.

dunefro commented on variables.tf:
```diff
@@ -116,6 +128,12 @@ variable "truefoundry_db_tier" {
 variable "truefoundry_db_zone" {
   type        = string
   description = "Zone for SQL DB - This must match the region"
+  default     = null
```
default should be empty string

dunefro commented on variables.tf:
```diff
@@ -51,7 +51,19 @@ variable "truefoundry_db_postgres_version" {
+
+  validation {
+    condition = (
+      var.truefoundry_db_enable ? (
```
this condition also has to change if the network CIDR is using default value

sachincool committed:
fix(variables.tf): update CIDR validation and defaults for clarity and correctness

Change the minimum CIDR range from /24 to /28 for better subnet allocation.
Set default CIDR to an empty string to enforce explicit user input.
Update error message to reflect new CIDR range.
Change default zone value to an empty string for consistency.

github-actions[bot] committed:
terraform-docs: automated action
